pyroomacoustics/room.py

```diff
@@ -1808,28 +1809,32 @@ def __init__(self,
         if materials is not None:
 
             if absorption_compatibility_request:
-                import warnings
                 warnings.warn(
-                        'Because materials were specified, deprecated absorption parameter is ignored.',
+                        'Because `materials` were specified, deprecated '
+                        '`absorption` parameter is ignored.',
                         DeprecationWarning,
                         )
 
             if isinstance(materials, Material):
                 materials = dict(zip(self.wall_names, [materials] * n_walls))
-            elif not isinstance(materials, dict):
-                raise ValueError('Materials must be a string, Material object, or dictionary')
+            else:
+                raise ValueError('`materials` must be a `Material` object.')
```

Comment:
Why not keep the dictionary ? I think it can be good if we want to set different materials for walls.

Reply:
ah sorry, I thought it was a "leftover" like the string! but now I see later on you were creating a dict for all cases to pass to libroom. I'll put it back and also add some docstring :)